src/cdk-experimental/dialog/dialog-container.ts

```diff
@@ -241,34 +245,72 @@ export class CdkDialogContainer extends BasePortalOutlet implements OnDestroy {
   }
 
   /**
-   * Autofocus the first tabbable element inside of the dialog, if there is not a tabbable element,
-   * focus the dialog instead.
+   * Focuses the provided element. If the element is not focusable, it will add a tabIndex
+   * attribute to forcefully focus it. The attribute is removed after focus is moved.
+   * @param element The element to focus.
    */
-  private _autoFocusFirstTabbableElement() {
+  private _forceFocus(element: HTMLElement, options?: FocusOptions) {
+    if (!this._interactivityChecker.isFocusable(element)) {
```

Comment:
Instead of checking if the element is focusable, we could just call `element.focus()`, then check to see if `element === document.activeElement`. If not, then apply `tabindex = "-1"` and focus again. WDYT?

Reply:
That makes sense. Although it seems strange to have different functionality for the same thing. @jelbourn do you have thoughts on this?

Reply:
I would probably stick with using `InteractivityChecker` _mostly_ for consistency, but also because I vaguely recall that Firefox will sometimes(?) not actually focus an element until the next microtask tick.